sample-apps/prometheus/main.go

```diff
@@ -0,0 +1,54 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"math/rand"
+	"net/http"
+	"time"
+
+	"log"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+var (
+	promRegistry = prometheus.NewRegistry() // local Registry so we don't get Go metrics, etc.
+)
+
+type metricBatch struct {
+	counter   prometheus.Counter
+	gauge     prometheus.Gauge
+	histogram prometheus.Histogram
+	summary   prometheus.Summary
+}
+
+func main() {
+	mc := newMetricCollector()
+
+	addressPtr := flag.String("listen_address", "0.0.0.0:8080", "server listening address")
```

Comment:
Alternatively you can do, 

```
flag.StringVar(&address, ...)
```

So you don't have to deal with the pointer variable names.

Reply:
Btw, feel free to rename this to "listen". That name is commonly used for this purpose and is shorter to type.